src/app/(user)/new_chat/[userId]/hooks/useChat.ts

```diff
@@ -15,18 +15,18 @@ export const useChat = ({ userId, socket }: Props) => {
   const { data, error } = useChatByUserId(userId);
   useEffect(() => {
     if (data) {
-      if (data.user?.userId !== userId) {
-        socket.emit("create");
-        socket.on("chatCreated", (chat: Chat) => {
-          socket.emit("join", { chatId: chat.chatId });
+      if (data.user?.id !== userId) {
+        socket.emit('create');
+        socket.on('chatCreated', (chat: Chat) => {
+          socket.emit('join', { chatId: chat.chatId });
           setChatId(chat.chatId);
         });
         return;
       }
-      if (data.chatId) {
-        console.log("Chat exists");
-        socket.emit("join", { chatId: data.chatId });
-        setChatId(data.chatId);
+      if (data.id) {
+        console.log('Chat exists');
+        socket.emit('join', { chatId: data.id });

```

Comment:
if you change the "chatId" or "userId" across all the app to "id",
please be consistent, change everywhere, in the socket handle as well as here and across everywhere where needed.

Reply:
in here it is not mandatory because you have defined the chat type in this file, I dont understand why you did it that way but you probably had your reassons. anyway I did it

Reply:
also I have added an anotation of `data : chat` because it is more explainatory that way